src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/BOM/BOMTable.tsx

```diff
@@ -101,6 +104,21 @@ const BOMTable: React.FC<BOMTableProps> = ({ columns, materials, assemblies }) =
       }}
     >
       <BomStyledDataGrid
+        onColumnVisibilityModelChange={(model: GridColumnVisibilityModel) => {
```

Comment:
Can you explain what this line does? I tried to run this in my local but I wasn't able to trigger the code of setColumn under the if statement on line 116. 

Alternatively, to reduce the complexity of the code here, why don't we call setColumn only when the user clicks on the hide button that is in three dots next to the column title?

Reply:
Hey! that seems to be one of the fixes I tried to make because of an infinite recursion loop I was running into, but it looks like it wasn't working and I forgot to take it out. Also I believe that the changes are being made only when a column is hidden with onColumnVisibilityModelChange, The problem I'm having is that it seems the setColumn isn't setting hideColumn back in BOMTableWrapper. Am I potentially using it wrong?